### PR TITLE
Check intermediate result of metricsMgrTmasterLatch in stmgr unit test

### PR DESCRIPTION
**Pull request by huijunwu**
Upgrade CountDownLatch wait()

### CODE REVIEW

File: heron/common/src/cpp/threads/spcountdownlatch.cpp [hunk 1]
```diff
@@ -20,13 +20,20 @@ CountDownLatch::CountDownLatch(sp_uint32 count) : count_(count) {}
...
 
-void CountDownLatch::wait() {
+bool CountDownLatch::wait(sp_uint32 target, const std::chrono::seconds& d) {
   std::unique_lock<std::mutex> m(mutex_);
-  // If count is greater than 0, then wait until it is 0.
+  // If count is greater than target, then wait until it is target.
   // Else return immediately.
-  while (count_ > 0) {
-    cond_.wait(m);
+  if (d == d.zero()) {
```
Comment (objmagic) [inline heron/common/src/cpp/threads/spcountdownlatch.cpp]: can we use `std::chrono::seconds::zero()` here?

Comment (huijunw) [inline heron/common/src/cpp/threads/spcountdownlatch.cpp]: done

File: heron/common/src/cpp/threads/spcountdownlatch.cpp [hunk 2]
```diff
@@ -20,13 +20,20 @@ CountDownLatch::CountDownLatch(sp_uint32 count) : count_(count) {}
 
 CountDownLatch::~CountDownLatch() {}
 
-void CountDownLatch::wait() {
+bool CountDownLatch::wait(sp_uint32 target, const std::chrono::seconds& d) {
   std::unique_lock<std::mutex> m(mutex_);
-  // If count is greater than 0, then wait until it is 0.
+  // If count is greater than target, then wait until it is target.
```
Comment (kramasamy) [inline heron/common/src/cpp/threads/spcountdownlatch.cpp]: I am assuming d is duration - can make it a descriptive name?

Comment (huijunw) [inline heron/common/src/cpp/threads/spcountdownlatch.cpp]: done

**Comment by objmagic** [comment]
ping @kramasamy. Can you give an approval?

**Pull request merged**